game-core/src/main/java/games/strategy/engine/framework/startup/login/HmacSha512Authenticator.java

```diff
@@ -41,31 +41,36 @@ private HmacSha512Authenticator() {}
    *         node; never {@code null}.
    */
   static Map<String, String> newChallenge() {
+    // This length is relatively large for a nonce simply because we do not currently expire nonces after they have been
+    // issued. If this were done, and a short nonce lifetime was used, we could probably get away with 8 bytes of random
+    // data safely as long as we included the timestamp as part of the nonce.
+    //
+    // https://security.stackexchange.com/questions/1952/how-long-should-a-random-nonce-be
+    final int nonceLengthInBytes = 64;
+
+    // Salt length should match hash output size.
+    //
+    // https://crackstation.net/hashing-security.htm#salt
+    final int saltLengthInBytes = 64;
+
     return Maps.newHashMap(ImmutableMap.<String, String>builder()
-        .put(encodeProperty(ChallengePropertyNames.NONCE, newRandomBytes()))
-        .put(encodeProperty(ChallengePropertyNames.SALT, newRandomBytes()))
+        .put(encodeProperty(ChallengePropertyNames.NONCE, newRandomBytes(nonceLengthInBytes)))
+        .put(encodeProperty(ChallengePropertyNames.SALT, newRandomBytes(saltLengthInBytes)))
         .build());
   }
 
   private static Map.Entry<String, String> encodeProperty(final String name, final byte[] value) {
     return Maps.immutableEntry(name, Base64.getEncoder().encodeToString(value));
   }
 
-  private static byte[] newRandomBytes() {
+  private static byte[] newRandomBytes(final int length) {
     // It is sufficient to use the default non-strong secure PRNG for the platform because the secrets we are protecting
```

Comment:
This line is confusing, we are using a 'long-lived' nonce, but here we say the values are short-lived. While there is nuance that a nonce is one-time, the documentation does not seem to agree with itself. To understand a lot of the commentary here, I think one would already need to understand the underlying code at a pretty deep level. I think this indicates the documentation has a wrong audience, or KISS problem.